network/tests/topology.rs

```diff
@@ -190,87 +190,30 @@ async fn star_converges_to_mesh() {
         max_peers: MAX_PEERS,
         ..Default::default()
     };
-    let mut nodes = test_nodes(N, setup).await;
+
+    // A bootnode will be necessary at the center of the star for peers to get propagated.
+    let hub_setup = TestSetup {
+        is_bootnode: true,
+        ..setup.clone()
+    };
+
+    // Create the regular nodes.
+    let mut nodes = test_nodes(N - 1, setup).await;
+
+    // Insert the bootnode at the head of the list.
+    nodes.insert(0, test_node(hub_setup).await);
+
     connect_nodes(&mut nodes, Topology::Star);
     start_nodes(&nodes).await;
 
-    wait_until!(15, network_density(&nodes) >= 0.1, 200);
+    wait_until!(15, network_density(&nodes) >= 0.2, 200);
     wait_until!(
         15,
         degree_centrality_delta(&nodes) <= (MAX_PEERS - MIN_PEERS).into(),
         200
     );
 }
 
-#[tokio::test(flavor = "multi_thread")]
-#[ignore]
-async fn binary_star_contact() {
```

Comment:
why is this test removed? was it flaky or are other tests sufficient to test mesh formation?

Reply:
It was flaky, at least in its current state. I was planning on exploring cases similar in complexity once this is merged, I  may well re-introduce something similar to it. This was mostly to clean things up in preparation for the next set of changes. Happy to leave it in as an ignored case for now, if that's preferable.

Reply:
yeah, you can just add a comment why it's currently ignored

Reply:
I've reintroduced the test, with a few tweaks it's actually more stable than it previously was.